src/registerListeners.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionContext, TextDocument, workspace } from 'vscode';
+import { ext } from './extensionVariables';
+
+let lastUploadTime: number = 0;
+const hourInMilliseconds = 1000 * 60 * 60;
+
+export function registerListeners(ctx: ExtensionContext): void {
+    ctx.subscriptions.push(workspace.onDidSaveTextDocument(onDidSaveTextDocument));
+}
+
+function onDidSaveTextDocument(doc: TextDocument): void {
+    // If it's not a Dockerfile, or last upload time is within an hour, skip
+    if (doc.languageId !== 'dockerfile' || lastUploadTime + hourInMilliseconds > Date.now()) {
+        return;
+    }
+
+    lastUploadTime = Date.now();

```

Comment:
Nit: capture `Date.now()` and use the same value in both places.

Reply:
I wonder if an allocation wouldn't be more expensive?